core/src/sumeragi/mod.rs

```diff
@@ -246,9 +246,38 @@ impl Sumeragi {
     pub fn initialize_and_start_thread(
         sumeragi: Arc<Self>,
         genesis_network: Option<GenesisNetwork>,
+        block_hashes: &[HashOf<VersionedCommittedBlock>],
     ) -> ThreadHandler {
         let wsv = sumeragi.wsv_mutex_access().clone();
 
+        for (block_hash, i) in block_hashes
+            .iter()
+            .take(block_hashes.len().saturating_sub(1))
+            .zip(1u64..)
+        {
+            let block_height: u64 = i;
+            let block_ref = sumeragi.internal.kura.get_block_by_height(block_height).expect("Sumeragi could not load block that was reported as present. Please check that the block storage was not disconnected.");
+            assert_eq!(
+                block_ref.hash(),
+                *block_hash,
+                "Kura init correctly reported the block hash."
```

Comment:
isn't this reported if assertion fails? in which case the message should be negated, no?

Reply:
I would commend the inclusion of the message though.

Reply:
I think normally the assert message describes what is being asserted. Such that it then says, assertion failed: "Thing has correct value".